tests/arch/forte_stringFunctions_test.cpp

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Samator Indo Gas
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Ketut Kumajaya - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+#include <boost/test/unit_test.hpp>
+
+#include "../../../src/arch/forte_stringFunctions.h"
+
+BOOST_AUTO_TEST_SUITE (forte_stringFunctions_test)
+
+  BOOST_AUTO_TEST_CASE(wstringToString)
+  {
+    // "4diac 4 ever!" in Japanese
+#if defined(_MSC_VER) || defined(__MINGW32__)
+    const std::string testString = "4diac 4 \xe3\x81\xaf\xe4\xbb\x8a\xe3\x81\xbe\xe3\x81\xa7\xef\xbc\x81"; // u8"4diac 4 は今まで！"
+    const std::wstring resultString = L"4diac 4 \xe3\x81\xaf\xe4\xbb\x8a\xe3\x81\xbe\xe3\x81\xa7\xef\xbc\x81"; // L"4diac 4 は今まで！"
+#else
+    // Set the new and save the previous locale
+    // BUG: trigger std::runtime_error: locale::facet::_S_create_c_locale name not valid with musl
+    std::locale prev_loc = std::locale::global(std::locale("C.UTF-8"));
+    const std::string testString = u8"4diac 4 は今まで！"; // make sure to use UTF-8 text editor with Unicode support
+    const std::wstring resultString = L"4diac 4 は今まで！";
+#endif
+#if 0
+    std::cout << "wstringToString: " << testString.c_str() << std::endl;
+    std::cout << "wstringToString: " << forte_wstringToString(resultString).c_str() << std::endl << std::endl;
+#endif
+
+    BOOST_CHECK_EQUAL(testString.length(), forte_wstringToString(resultString).length());
```

Comment:
as a suggestion, I nowadays only use BOOST_TEST, as it usually results in better output if a test fails, like getting the exact numbers or strings.
You can also check C-strings directly for equality (as compare) when using BOOST_TEST
https://www.boost.org/doc/libs/1_85_0/libs/test/doc/html/boost_test/testing_tools/extended_comparison/strings.html